### PR TITLE
[System.IO] Use CoreFX FileSystemWatcher on Windows

### PR DESCRIPTION
**Pull request by alexischr**
Outstanding Issues:

- [x] https://jenkins.mono-project.com/job/x/8254/testReport/junit/MonoTests/runtime/bug_80307_exe/
- [ ] Fix System.Web failures

#6966

### CODE REVIEW

**Comment by alexischr** [comment]
@monojenkins build Windows i386

**Comment by alexischr** [comment]
@monojenkins build Windows x64

**Comment by alexischr** [comment]
@monojenkins build failed

File: mcs/class/System/System.IO/FileSystemWatcher.cs [hunk 1]
```diff
@@ -125,8 +125,8 @@ void InitWatcher ()
 				bool ok = false;
 				switch (mode) {
 				case 1: // windows
-					ok = DefaultWatcher.GetInstance (out watcher);
-					watcher_handle = this;
+					ok = CoreFXFileSystemWatcherProxy.GetInstance (out watcher);
```
Comment (Therzok) [inline mcs/class/System/System.IO/FileSystemWatcher.cs]: Wouldn't it be simpler to have:
```
case 1:
case 6:
 	ok = CoreFXFileSystemWatcherProxy.GetInstance (out watcher);
 	watcher_handle = (watcher as CoreFXFileSystemWatcherProxy).NewWatcher (this);
 	break;
```

Comment (alexischr) [reply mcs/class/System/System.IO/FileSystemWatcher.cs]: Sounds good :)

**Comment by alexischr** [comment]
@monojenkins build failed

**Comment by alexischr** [comment]
@monojenkins build failed

**Comment by alexischr** [comment]
@monojenkins build

**Comment by marek-safar** [comment]
Can this be now closed?

**Comment by alexischr** [comment]
@monojenkins build failed

**Comment by steveisok** [comment]
Since this is older, I'm closing.  Please see the [old pull request policy](https://github.com/mono/mono/wiki/Pull-Request-Guide#old-pull-request-policy) for more info.

**Pull request closed without merging**